src/query.ts

```diff
@@ -201,24 +207,29 @@ class Query {
    * const keyQuery = query.filter('__key__', key);
    * ```
    */
-  filter(property: string, value: {} | null): Query;
-  filter(property: string, operator: Operator, value: {} | null): Query;
+  filter(propertyOrFilter: string | NewFilter, value?: {} | null): Query;
```

Comment:
why can't this just be `string | Filter`?

Reply:
In this module, `NewFilter` is just what we call the imported `Filter` class. We cannot call it `Filter` because the name `Filter` is already taken in this module.

https://github.com/googleapis/nodejs-datastore/pull/1061/files#diff-ad0978c0145d3620d2a97bdff4307f9496529c0e3011a845e2d5bf1a169d2538R21

Reply:
Filter has been renamed `entityFilter` so as not to cause confusion.